detect/detect_test.go

```diff
@@ -855,6 +848,225 @@ func TestDetectWithSymlinks(t *testing.T) {
+					File:        "package.json",
+					Entropy:     3.095795154571533,
+					RuleID:      "test-rule",
+				},
+			},
+		},
+		"commit AND path NOT allowed - other conditions": {
+			fragment: Fragment{
+				CommitSHA: "41edf1f7f612199f401ccfc3144c2ebd0d7aeb48",
+				FilePath:  "package-lock.json",
+			},
+			allowlist: config.Allowlist{
+				MatchCondition: config.AllowlistMatchAnd,
+				Commits:        []string{"41edf1f7f612199f401ccfc3144c2ebd0d7aeb48"},
+				Paths:          []*regexp.Regexp{regexp.MustCompile(`package-lock.json`)},
+				Regexes:        []*regexp.Regexp{regexp.MustCompile("password")},
+			},
+			expected: []report.Finding{
+				{
+					StartColumn: 50,
+					EndColumn:   60,
+					Line:        "let username = 'james@mail.com';\nlet password = 'Summer2024!';",
+					Match:       "Summer2024!",
+					Secret:      "Summer2024!",
+					File:        "package-lock.json",
+					Entropy:     3.095795154571533,
+					RuleID:      "test-rule",
+				},
+			},
+		},
+		"commit OR path allowed": {
+			fragment: Fragment{
+				CommitSHA: "41edf1f7f612199f401ccfc3144c2ebd0d7aeb48",
+				FilePath:  "package-lock.json",
+			},
+			allowlist: config.Allowlist{
+				MatchCondition: config.AllowlistMatchOr,
+				Commits:        []string{"704178e7dca77ff143778a31cff0fc192d59b030"},
+				Paths:          []*regexp.Regexp{regexp.MustCompile(`package-lock.json`)},
+			},
+		},
+
+		// Regex / stopwords
+		"regex allowed": {
+			fragment: Fragment{},
+			allowlist: config.Allowlist{
+				Regexes: []*regexp.Regexp{regexp.MustCompile(`(?i)summer.+`)},
+			},
+		},
+		"stopwords allowed": {
+			fragment: Fragment{},
+			allowlist: config.Allowlist{
+				StopWords: []string{"summer"},
+			},
+		},
+		"regex AND stopword allowed": {
+			fragment: Fragment{},
+			allowlist: config.Allowlist{
+				MatchCondition: config.AllowlistMatchAnd,
+				Regexes:        []*regexp.Regexp{regexp.MustCompile(`(?i)summer.+`)},
+				StopWords:      []string{"2024"},
+			},
+		},
+		"regex AND stopword allowed - other conditions": {
+			fragment: Fragment{
+				CommitSHA: "41edf1f7f612199f401ccfc3144c2ebd0d7aeb48",
+				FilePath:  "config.js",
+			},
+			allowlist: config.Allowlist{
+				MatchCondition: config.AllowlistMatchAnd,
+				Commits:        []string{"41edf1f7f612199f401ccfc3144c2ebd0d7aeb48"},
+				Paths:          []*regexp.Regexp{regexp.MustCompile(`config.js`)},
+				Regexes:        []*regexp.Regexp{regexp.MustCompile(`(?i)summer.+`)},
+				StopWords:      []string{"2024"},
+			},
+		},
+		"regex AND stopword allowed - non-git, other conditions": {
+			fragment: Fragment{
+				FilePath: "config.js",
+			},
+			allowlist: config.Allowlist{
+				MatchCondition: config.AllowlistMatchAnd,
+				Commits:        []string{"41edf1f7f612199f401ccfc3144c2ebd0d7aeb48"},
+				Paths:          []*regexp.Regexp{regexp.MustCompile(`config.js`)},
+				Regexes:        []*regexp.Regexp{regexp.MustCompile(`(?i)summer.+`)},
+				StopWords:      []string{"2024"},
+			},
```

Comment:
If this is a non-git scan, should this match on paths+regexes+stopwords? Or should allowlists that include `commits` it ONLY match git scans?

Reply:
stopwords have always been independent of `regex` and `path`. I think stopwords should continue to have the highest authority on ignoring a finding. 

> Or should AND allowlists that include commits it ONLY match git scans?

I would say only match git scans since the default config does not include any commit fields in rule allowlists. If youre going through the trouble of configuring a rule you could just copy and paste your existing rule w/ and w/o a commit field in the allowlist. I hope that makes sense.